src/erc721/ERC721.sol

```diff
@@ -0,0 +1,237 @@
+                keccak256("EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)"),
+                keccak256(bytes(name)),
+                keccak256(bytes("1")),
+                block.chainid,
+                address(this)
+            )
+        );
+    }
+    
+    /*///////////////////////////////////////////////////////////////
+                              ERC-20-LIKE LOGIC
+    //////////////////////////////////////////////////////////////*/
+    
+    function transfer(address to, uint256 tokenId) external {
+        require(msg.sender == ownerOf[tokenId], "NOT_OWNER");
+        
+        // This is safe because ownership is checked
+        // against decrement, and sum of all user
+        // balances can't exceed type(uint256).max!
+        unchecked {
+            balanceOf[msg.sender]--; 
+        
+            balanceOf[to]++;
+        }
+        
+        delete getApproved[tokenId];
+        
+        ownerOf[tokenId] = to;
+        
+        emit Transfer(msg.sender, to, tokenId); 
+    }
+    
+    /*///////////////////////////////////////////////////////////////
+                              ERC-721 LOGIC
+    //////////////////////////////////////////////////////////////*/
+    
+    function supportsInterface(bytes4 interfaceId) external pure returns (bool supported) {
+        supported = interfaceId == 0x80ac58cd || interfaceId == 0x5b5e139f;
+    }
+    
+    function approve(address spender, uint256 tokenId) external {
+        address owner = ownerOf[tokenId];
+        
+        require(msg.sender == owner || isApprovedForAll[owner][msg.sender], "NOT_APPROVED");
+        
+        getApproved[tokenId] = spender;
+        
+        emit Approval(owner, spender, tokenId); 
+    }
+    
+    function setApprovalForAll(address operator, bool approved) external {
+        isApprovedForAll[msg.sender][operator] = approved;
+        
+        emit ApprovalForAll(msg.sender, operator, approved);
+    }
+
+    function transferFrom(address, address to, uint256 tokenId) public {
+        address owner = ownerOf[tokenId];
+        
+        require(
+            msg.sender == owner 
+            || msg.sender == getApproved[tokenId]
+            || isApprovedForAll[owner][msg.sender], 
+            "NOT_APPROVED"
+        );
+        
+        // This is safe because ownership is checked
+        // against decrement, and sum of all user
+        // balances can't exceed type(uint256).max!
+        unchecked { 
+            balanceOf[owner]--; 
+        
+            balanceOf[to]++;
+        }
+        
+        delete getApproved[tokenId];
+        
+        ownerOf[tokenId] = to;
+        
+        emit Transfer(owner, to, tokenId); 
+    }
+    
+    function safetransferFrom(address, address to, uint256 tokenId) external {
+        safetransferFrom(address(0), to, tokenId, "");
+    }
+    
+    function safetransferFrom(address, address to, uint256 tokenId, bytes memory data) public {
```

Comment:
Both safeTransferFrom have a typo on the T which should be uppercase

Reply:
Good catch. @Ro5s make sure to update TridentNFT too! :p

Reply:
lazer eyed @dievardump -- will make those changes :~0)